native/utils.h

```diff
@@ -9,8 +9,33 @@
 //   Utils
 // =============================================================================
 
-void vec_rect_cxx_to_c(const std::vector<cv::Rect>& cxx_vec_rect, VecRect* vr);
-void vec_double_cxx_to_c(const std::vector<double>& cxx_vec, VecDouble* v);
-void vec_point_cxx_to_c(const std::vector<cv::Point>& cxx_vec_point, VecPoint* vp);
-void vec_points_cxx_to_c(const std::vector<std::vector<cv::Point>> &cxx_vec_points, VecPoints* vps);
+void cv_to_ffi(const cv::Rect& source, Rect* dest){
```

Comment:
The implementation should be moved to `.cc` file. Otherwise, when `opencv-wrapper.c` includes `utils.h` and `opencv-gpu.cc` includes `utils.h`, we have a multiple definition issue.
That is, both `opencv-wrapper.o` and `opencv-gpu.o` have definitions of `cv_to_ffi` for the same argument.

Reply:
Yep, I tried it. However, templates cannot be moved out of `.h` file, see explanation: https://stackoverflow.com/questions/495021/why-can-templates-only-be-implemented-in-the-header-file.
You can add a commit yourself if you see how it can be done.

Reply:
Leave the template function in `.h` and move the implementation to `.cc`.

```
// utils.h
void cv_to_ffi(const cv::Rect& source, Rect* dest);
void cv_to_ffi(const cv::Point& source, Point2i* dest);
void cv_to_ffi(const std::vector<double>& source, CVec<double>* dest);

template <typename T, typename U>
void cv_to_ffi(const std::vector<T>& source, CVec<U>* dest)
{
    size_t num = source.size();
    dest->size = num;
    dest->array = (U*) malloc(num * sizeof(U));
    for (size_t i = 0; i < num; i++) {
        cv_to_ffi(source[i], &dest->array[i]);
    }
}

#endif  // UTILS_H_
```

Reply:
Mmm, hitting an error:

```
     Running target/debug/deps/test_objdetect-af8e9348272df2ef

running 2 tests
error: An unknown error occurred
```

Let me take a look.

Reply:
I did it, however I have a question: why double definition is not a problem for templates?

Reply:
Interesting, it's passing on Travis.

Reply:
I traced down the bug. See my updated comment (request for change).

Reply:
templates are not compiled to object files, see https://stackoverflow.com/questions/7584781/inclusion-of-header-files-in-case-of-templates